seatunnel-engine/seatunnel-engine-common/src/main/java/org/apache/seatunnel/engine/common/utils/ExceptionUtil.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.engine.common.utils;
+
+import org.apache.seatunnel.engine.common.exception.JobNotFoundExceptionSeaTunnel;
+import org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException;
+
+import com.hazelcast.client.impl.protocol.ClientExceptionFactory;
+import com.hazelcast.client.impl.protocol.ClientProtocolErrorCodes;
+import com.hazelcast.instance.impl.OutOfMemoryErrorDispatcher;
+import lombok.NonNull;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+
+public final class ExceptionUtil {
```

Comment:
We may need to add this class in LICENSE file. cc @CalvinKirs

Reply:
Hi, where does this class come from?

Reply:
This class is similarity with [ExceptionUtil](https://github.com/hazelcast/hazelcast-jet/blob/8e7c2995ec2c9d47a2057f018c85c89ef5cac609/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ExceptionUtil.java)

Reply:
I was a little worried, is there any other similar situation in the code of this PR? @EricJoy2048

Reply:
done

Reply:
we need to publish the binary package, so, please also add to dist/NOTICE.
and is there any other similar situation in the code of this PR?

Reply:
> we need to publish the binary package, so, please also add to dist/NOTICE. and is there any other similar situation in the code of this PR?

I don't know `SeaTunnelEngineException` whether need to do this. @CalvinKirs  @ruanwenjun

Reply:
> we need to publish the binary package, so, please also add to dist/NOTICE. and is there any other similar situation in the code of this PR?

In hazelcast 5.1 hazelcast-jet merge to hazelcast, So the source code dependency is point to hazelcast.